### PR TITLE
New Extension: Currency exchange

### PR DESCRIPTION
**Pull request by xeric**
## Description

A new Currency exchange extension which provides better user experience.
See (README.md) for more details.

## Screencast

expression
filter

## Checklist

- [ √] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [√ ] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [ √] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [ √] I checked that files in the `assets` folder are used by the extension itself
- [ √] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

File: extensions/currency-exchange/package.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      "title": "Currency",
+      "subtitle": "currency exchange",
+      "description": "Calculate with Exchange Rate",
+      "mode": "view"
+    }
+  ],
+  "preferences": [
+    {
+      "name": "api_key",
+      "type": "textfield",
```
Comment (pernielsentikaer) [inline extensions/currency-exchange/package.json:21]:
```suggestion
      "type": "password",
```

File: extensions/currency-exchange/package.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  "title": "Currency Exchange",
+  "description": "Simple Currency Exchange",
+  "icon": "currency.png",
+  "author": "xeric",
+  "license": "MIT",
+  "commands": [
+    {
+      "name": "index",
+      "title": "Currency",
+      "subtitle": "currency exchange",
```
Comment (pernielsentikaer) [inline extensions/currency-exchange/package.json:13]:
```suggestion
      "title": "Currency Exchange",
```

File: extensions/currency-exchange/src/index.tsx [hunk 3]
```diff
@@ -0,0 +1,397 @@
...
+        }));
+      }
+    })();
+  }, []);
+
+  return (
+    <List
+      isLoading={state.isLoading}
+      onSearchTextChange={exchange}
+      searchBarPlaceholder={`input how much ${currencyCode2Name[state.fromCurrencyCode]}`}
```
Comment (pernielsentikaer) [inline extensions/currency-exchange/src/index.tsx:36]:
```suggestion
      searchBarPlaceholder={`Input how much ${currencyCode2Name[state.fromCurrencyCode]}`}
```

File: extensions/currency-exchange/src/index.tsx [hunk 4]
```diff
@@ -0,0 +1,397 @@
...
+            LocalStorage.setItem(STORAGE_KEY_PINNED_CURRENCY_CODE, JSON.stringify(pinnedUpdated));
+            setState((oldState) => ({
+              ...oldState,
+              pinnedCurrencyCodes: pinnedUpdated,
+            }));
+          }
+        }}
+        shortcut={{ modifiers: ["shift"], key: "return" }}
+        title={`${
+          state.pinnedCurrencyCodes && state.pinnedCurrencyCodes.indexOf(toCurrencyCode) >= 0 ? "UnPin It" : "Pin It"
```
Comment (pernielsentikaer) [inline extensions/currency-exchange/src/index.tsx:176]:
```suggestion
          state.pinnedCurrencyCodes && state.pinnedCurrencyCodes.indexOf(toCurrencyCode) >= 0 ? "Unpin It" : "Pin It"
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @xeric 👋 

Thanks for another contribution 🔥 

I left some feedback and suggestions for you:

- When using wrong credentials I get this, is it possible to format the text a bit more understandable?
image

- When entering the extension I get this error in the console, could you ensure that keys are set correctly?
```
09:10:50.274 Warning: Each child in a list should have a unique "key" prop.

Check the render method of `Command`. See https://reactjs.org/link/warning-keys for more information.
    at Tu (/Applications/Raycast.app/Contents/Resources/RaycastCommands_RaycastCommands.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:24:27807)
    at null.Command (/Users/pernielsentikaer/Documents/Raycast/tmp/currency-exchange/src/index.tsx:10:41)
    at Kr (/Applications/Raycast.app/Contents/Resources/RaycastCommands_RaycastCommands.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:24:2463)
    at ray-navigation-stack
    at No (/Applications/Raycast.app/Contents/Resources/RaycastCommands_RaycastCommands.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:24:2091)
    at ko (/Applications/Raycast.app/Contents/Resources/RaycastCommands_RaycastCommands.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:23:619)
    at ray-root
    at ba (/Applications/Raycast.app/Contents/Resources/RaycastCommands_RaycastCommands.bundle/Contents/Resources/api/node_modules/@raycast/api/index.js:24:2832)
```

- Could you rename the folder `images` to `media`

- Maybe add a cool [List.EmptyView](https://developers.raycast.com/api-reference/user-interface/list#list.emptyview) for initial awesomeness 🔥 

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

**Comment by xeric** [comment]
@pernielsentikaer I've done all suggested changes, please help to review them, thanks.

File: extensions/currency-exchange/package.json [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      "title": "API Key",
+      "description": "You can get API Key at https://app.exchangerate-api.com/dashboard",
+      "placeholder": "Exchange Rate API Key"
+    }
+  ],
+  "dependencies": {
+    "@raycast/api": "^1.35.0",
+    "node-fetch": "^3.0.0",
+    "run-applescript": "^6.0.0",
+    "sound-play": "^1.1.0"
```
Comment (pernielsentikaer) [inline extensions/currency-exchange/package.json:31]: Are they even required?

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @xeric 👋 

Thanks a lot for the update 🔥 

I added changelog and metadata images for you 💪 I have one last question before merging, found some possible leftovers in the package.json that we should remove if not needed

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/currency-exchange/package.json (hunk 5 above)
Comment (xeric) [reply extensions/currency-exchange/package.json:31]: Great catch! I forget to clean up useless dependencies, will update ASAP 👍

**Comment by pernielsentikaer** [comment]
Much better @xeric, lets's get it merged! 🔥

**Pull request merged**